addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -163,6 +163,12 @@ def _update_workorder_lines(self):
                 if float_compare(qty_todo, 0.0, precision_rounding=rounding) > 0:
                     for values in self._generate_lines_values(move, qty_todo):
                         line_values['to_create'].append(values)
+        # wo lines without move_id should also be deleted
+        for wo_line in self.raw_workorder_line_ids.filtered(lambda w: not w.move_id):
```

Comment:
I guess the same issue could come if you remove a byproduct from the BoM. I would also loop on the finished_workorder_line_ids (you can get both recordset via _workorder_line_ids() ). 
Pay attention to not remove workorder lines that refer to the finished product as they serve another purpose and don't have any stock move attached

Reply:
Thanks for the feedback.
At the beginning, I was looping on both as you are suggesting, using:
`for wo_line in self._workorder_line_ids().filtered(lambda w: not w.move_id):`
But some tests were failing: https://runbot.odoo.com/runbot/build/4658585

Maybe it is linked to your remark.

> Pay attention to not remove workorder lines that refer to the finished product as they serve another purpose and don't have any stock move attached

Reply:
Those lines store lot_id for the finished product. The failing tests seem related to that topic :)

Reply:
Is there a way to differentiate those lines from byproduct ones ?

Reply:
The product_id is the one you produce on the manufacturing order.

Reply:
Ah ok. Thanks.